### PR TITLE
fix: Remove target availability event listener on dispose (#29)

### PR DESCRIPTION
**Pull request by izkmdz**
Fix for issue #29.

This commit removes the "webkitplaybacktargetavailabilitychanged" event listener when the component is disposed. It also provides a test case demo in which the user can remove (dispose) the AirPlay component from the videoJS player.

### CODE REVIEW

File: docs/demo/test-cases/remove-air-play-button.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+   <button id="removeBtn" onclick="removeAirPlayButton()">Remove AirPlay Button</button>
+
+   <script>
+      var player = videojs("video_1", {}, function() {
+         player.airPlay();
+      });
+
+      removeAirPlayButton = function() {
+         var airPlayButton = player.controlBar.getChild('airPlayButton');
+         airPlayButton.dispose();
```
Comment (yokuze) [inline docs/demo/test-cases/remove-air-play-button.html:40]: We don't have linting in HTML files, so the linter wouldn't have caught this: our coding standards have us separate `var` statements from the lines that follow using a newline.

File: src/js/components/AirPlayButton.js [hunk 2]
```diff
@@ -89,12 +89,17 @@ AirPlayButton = {
          return;
       }
 
-      mediaEl.addEventListener('webkitplaybacktargetavailabilitychanged', function(event) {
+      function _onTargetAvailabilityChanged(event) {
```
Comment (yokuze) [inline src/js/components/AirPlayButton.js]: Just a minor nit: we don't normally start variable/function/method names with an underscore unless they are private or protected.

File: docs/demo/test-cases/remove-air-play-button.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+         document.getElementById("removeBtn").disabled = true;
```
Comment (yokuze) [inline docs/demo/test-cases/remove-air-play-button.html]: "removeBtn" -> 'removeBtn'

File: docs/demo/test-cases/remove-air-play-button.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+   <h1>Demo of <code>silvermine-videojs-airplay</code></h1>
+
+   <video id="video_1" class="video-js vjs-default-skin" controls preload="auto" data-setup='{ "fluid": "true" }'>
+      <source src="http://www.caminandes.com/download/03_caminandes_llamigos_1080p.mp4" type="video/mp4">
+   </video>
+
+   <button id="removeBtn" onclick="removeAirPlayButton()">Remove AirPlay Button</button>
+
+   <script>
+      var player = videojs("video_1", {}, function() {
```
Comment (yokuze) [inline docs/demo/test-cases/remove-air-play-button.html]: "video_1" -> 'video_1'

File: docs/demo/test-cases/remove-air-play-button.html [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+<head>
+   <meta charset=utf-8 />
+   <title>silvermine-videojs-airplay Demo</title>
+   <link href="https://unpkg.com/video.js@7.4.1/dist/video-js.css" rel="stylesheet">
+   <script src="https://unpkg.com/video.js@7.4.1/dist/video.js"></script>
+   <script src="../../dist/silvermine-videojs-airplay.min.js"></script>
+   <link href="../../dist/silvermine-videojs-airplay.css" rel="stylesheet">
+</head>
+<body>
+   <h1>Demo of <code>silvermine-videojs-airplay</code></h1>
```
Comment (yokuze) [inline docs/demo/test-cases/remove-air-play-button.html]: Could we change the title and add a description below it that helps the reader to understand the purpose of the test case? Maybe something like:

`<a href="https://github.com/silvermine/videojs-airplay/issues/29>#29</a> Removing/disposing of the AirPlay button`

`When the AirPlay button is programmatically removed from the player, this plugin should remove the event listeners that it registered outside of its own scope (e.g. on the player's media element). If this test case is failing, you will see an error in the console when you click the "Remove AirPlay Button" button.`

**Review by yokuze — CHANGES_REQUESTED** [review]

File: src/js/components/AirPlayButton.js [hunk 6]
```diff
@@ -89,12 +89,17 @@ AirPlayButton = {
...
-      mediaEl.addEventListener('webkitplaybacktargetavailabilitychanged', function(event) {
+      function onTargetAvailabilityChanged(event) {
          if (event.availability === 'available') {
             self.show();
          } else {
             self.hide();
          }
+      }
+
+      mediaEl.addEventListener('webkitplaybacktargetavailabilitychanged', onTargetAvailabilityChanged, this);
```
Comment (yokuze) [inline src/js/components/AirPlayButton.js]: One other thing I missed in the previous review: The third parameter to the `addEventListener` method is either an options object or a boolean flag, not a reference to `this`: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener

We should just remove the third parameter here and below. The `onTargetAvailabilityChanged` handler has access to the proper `this` via the `self` variable.

Comment (izkmdz) [reply src/js/components/AirPlayButton.js]: Fixed. 👍  Thanks!

**Review by yokuze — APPROVED** [review]

**Pull request merged**